datadog_checks_base/tests/test_agent_check.py

```diff
@@ -45,3 +48,61 @@ def test_unicode_string(self):
 
         assert normalized_tags is not tags
         assert normalized_tag == tag.encode('utf-8')
+
+
+class LimitedCheck(AgentCheck):
+    DEFAULT_METRIC_LIMIT = 10
+
+
+class TestLimits(unittest.TestCase):
+    def tearDown(self):
+        aggregator.reset()
+
+    def test_metric_limit_gauges(self):
+        check = LimitedCheck()
+        assert check.get_warnings() == []
+
+        for i in range(0, 10):
+            check.gauge("metric", 0)
+        assert len(check.get_warnings()) == 0
+        assert len(aggregator.metrics("metric")) == 10
+
+        for i in range(0, 10):
+            check.gauge("metric", 0)
+        assert len(check.get_warnings()) == 1
+        assert len(aggregator.metrics("metric")) == 10
+
+    def test_metric_limit_count(self):
```

Comment:
That's a great test 💯 It covers the internal logic of the limiter very well, and provides an example that helps understanding how the Limiter works.